### PR TITLE
Deauthorize sortition pool- dapp

### PR DESCRIPTION
**Pull request by r-czajkowski**
See https://github.com/keep-network/keep-ecdsa/pull/471
Closes: #1837 

This PR allows an authorizer to deauthorize a `TBTCSystem sortition pool` for a given operator.

### CODE REVIEW

File: solidity/dashboard/src/pages/TBTCApplicationPage.jsx [hunk 1]
```diff
@@ -83,8 +83,8 @@ const TBTCApplicationPage = () => {
     unbondedValueWithdrawnCallback
   )
 
-  const onAuthorizationSuccessCallback = useCallback(
-    (contractName, operatorAddress) => {
+  const onSuccessCallback = useCallback(
+    (contractName, operatorAddress, isDeauthorization = false) => {
```
Comment (dimpar) [inline solidity/dashboard/src/pages/TBTCApplicationPage.jsx:87]: what do you think about removing `isDeauthorization` and replacing with `isAuthorized = true`?  We have only 2 cases: `isAuthorized = true / false` it seems that it should cover `isDeauthorization` as well.

File: solidity/dashboard/src/pages/TBTCApplicationPage.jsx [hunk 2]
```diff
@@ -108,7 +108,7 @@ const TBTCApplicationPage = () => {
       const updatedContracts = [...obsoleteOperator.contracts]
       updatedContracts[contractIndexInArray] = {
         ...obsoleteContract,
-        isAuthorized: true,
+        isAuthorized: !isDeauthorization,
```
Comment (dimpar) [inline solidity/dashboard/src/pages/TBTCApplicationPage.jsx:111]: depending on the prev. comment, we might just leave `isAuthorized`

File: solidity/dashboard/src/pages/TBTCApplicationPage.jsx [hunk 3]
```diff
@@ -139,20 +139,47 @@ const TBTCApplicationPage = () => {
...
+          transactionHashCallback
+        )
+        showMessage({
+          type: messageType.SUCCESS,
+          title: "Success",
+          content: "Deauthorization transaction successfully completed",
+        })
         setTimeout(
-          () => onAuthorizationSuccessCallback(contractName, operatorAddress),
+          () => onSuccessCallback("TBTCSystem", operatorAddress, true),
```
Comment (dimpar) [inline solidity/dashboard/src/pages/TBTCApplicationPage.jsx:170]: also here, depending on the comment regarding `isDeathorization` we might want to pass `false` here

File: solidity/dashboard/src/pages/TBTCApplicationPage.jsx [hunk 4]
```diff
@@ -139,20 +139,47 @@ const TBTCApplicationPage = () => {
           title: "Success",
           content: "Authorization transaction successfully completed",
         })
+        setTimeout(() => onSuccessCallback(contractName, operatorAddress), 5000)
+      } catch (error) {
+        showMessage({
+          type: messageType.ERROR,
+          title: "Authorization action has failed ",
```
Comment (dimpar) [inline solidity/dashboard/src/pages/TBTCApplicationPage.jsx:146]: not super strong on this, but what do you think about unifying notification messages for authorization/deathorization? maybe just simply:
- "Authorization successfully completed"
- "Authorization has failed"
- "Deauthorization successfully completed"
- "Deauthorization has failed"

File: solidity/dashboard/src/services/tbtc-authorization.service.js [hunk 5]
```diff
@@ -415,11 +415,39 @@ const fetchAvailableAmount = async (web3Context, operator) => {
...
+  const { keepBondingContract, yourAddress } = web3Context
+  const { operatorAddress, poolAddress } = data
+
+  await keepBondingContract.methods
+    .deauthorizeSortitionPoolContract(operatorAddress, poolAddress)
+    .send({ from: yourAddress })
+    .on("transactionHash", onTransactionHashCallback)
+}
+
+const deauthorizeTBTCSystem = async (
```
Comment (dimpar) [inline solidity/dashboard/src/services/tbtc-authorization.service.js:432]: what do you think about leaving just:
```
const deauthorizeTBTCSystem = async (
  web3Context,
  operatorAddress,
  onTransactionHashCallback
) => {
  const { keepBondingContract, yourAddress } = web3Context
  const poolAddress = await fetchSortitionPoolForTbtc(web3Context)

  await keepBondingContract.methods
    .deauthorizeSortitionPoolContract(operatorAddress, poolAddress)
    .send({ from: yourAddress })
    .on("transactionHash", onTransactionHashCallback)
}
```

Comment (r-czajkowski) [reply solidity/dashboard/src/services/tbtc-authorization.service.js:432]: I think we can leave `deauthorizeSortitionPoolContract` function for future.

Comment (dimpar) [reply solidity/dashboard/src/services/tbtc-authorization.service.js:432]: do we already have a case in mind for `deauthorizeSortitionPoolContract` function? If yes, then lets leave it, if no, YAGNI principle says to remove it :)

**Pull request merged**